mirascope/core/litellm/_utils/_setup_call.py

```diff
@@ -25,22 +69,23 @@ def setup_call(
     call_params: OpenAICallParams,
     extract: bool,
 ) -> tuple[
-    Callable[..., ChatCompletion] | Callable[..., Awaitable[ChatCompletion]],
-    str,
+    Callable[..., ModelResponse | CustomStreamWrapper]
```

Comment:
hmm, even though this is technically correct, we should type this as returning the OpenAI types because for all intents and purposes that's how we're treating them.

To remove the type ignore on the return type below, we should just cast the function to match the type?

Reply:
I just reverted it